tests/ecdsa-common.c

```diff
@@ -0,0 +1,2849 @@
+      s[1] =  0x444e323075d98f31ul;
+      s[0] =  0x920b9e0f57ec871cul;
+
+      bool result = Hacl_Impl_ECDSA_ecdsa_p256_sha2_verify (mLen, m, pubKey, r, s);
+
+      bool expectedResult = true;
+      if (result == expectedResult)
+        printf("%s\n", "Test0: passed");
+      else
+        printf("%s\n", "Test0: failed");
+}
+
+/*
+Msg = e4796db5f785f207aa30d311693b3702821dff1168fd2e04c0836825aefd850d9aa60326d88cde1a23c7745351392ca2288d632c264f197d05cd424a30336c19fd09bb229654f0222fcb881a4b35c290a093ac159ce13409111ff0358411133c24f5b8e2090d6db6558afc36f06ca1f6ef779785adba68db27a409859fc4c4a0
+Qx = 87f8f2b218f49845f6f10eec3877136269f5c1a54736dbdf69f89940cad41555
+Qy = e15f369036f49842fac7a86c8a2b0557609776814448b8f5e84aa9f4395205e9
+R = d19ff48b324915576416097d2544f7cbdf8768b1454ad20e0baac50e211f23b0
+S = a3e81e59311cdfff2d4784949f7a2cb50ba6c3a91fa54710568e61aca3e847c6
+Result = F (3 - S changed)
+*/
+
+void testEcdsaVerification2()
+   {
+      uint8_t* m = (uint8_t *) malloc (sizeof (uint8_t) * 128);
+      m[0] = 0xe4;
+      m[1] = 0x79;
+      m[2] = 0x6d;
+      m[3] = 0xb5;
+      m[4] = 0xf7;
+      m[5] = 0x85;
+      m[6] = 0xf2;
+      m[7] = 0x07;
+      m[8] = 0xaa;
+      m[9] = 0x30;
+      m[10] = 0xd3;
+      m[11] = 0x11;
+      m[12] = 0x69;
+      m[13] = 0x3b;
+      m[14] = 0x37;
+      m[15] = 0x02;
+      m[16] = 0x82;
+      m[17] = 0x1d;
+      m[18] = 0xff;
+      m[19] = 0x11;
+      m[20] = 0x68;
+      m[21] = 0xfd;
+      m[22] = 0x2e;
+      m[23] = 0x04;
+      m[24] = 0xc0;
+      m[25] = 0x83;
+      m[26] = 0x68;
+      m[27] = 0x25;
+      m[28] = 0xae;
+      m[29] = 0xfd;
+      m[30] = 0x85;
+      m[31] = 0x0d;
+      m[32] = 0x9a;
+      m[33] = 0xa6;
+      m[34] = 0x03;
+      m[35] = 0x26;
+      m[36] = 0xd8;
+      m[37] = 0x8c;
+      m[38] = 0xde;
+      m[39] = 0x1a;
+      m[40] = 0x23;
+      m[41] = 0xc7;
+      m[42] = 0x74;
+      m[43] = 0x53;
+      m[44] = 0x51;
+      m[45] = 0x39;
+      m[46] = 0x2c;
+      m[47] = 0xa2;
+      m[48] = 0x28;
+      m[49] = 0x8d;
+      m[50] = 0x63;
+      m[51] = 0x2c;
+      m[52] = 0x26;
+      m[53] = 0x4f;
+      m[54] = 0x19;
+      m[55] = 0x7d;
+      m[56] = 0x05;
+      m[57] = 0xcd;
+      m[58] = 0x42;
+      m[59] = 0x4a;
+      m[60] = 0x30;
+      m[61] = 0x33;
+      m[62] = 0x6c;
+      m[63] = 0x19;
+      m[64] = 0xfd;
+      m[65] = 0x09;
+      m[66] = 0xbb;
+      m[67] = 0x22;
+      m[68] = 0x96;
+      m[69] = 0x54;
+      m[70] = 0xf0;
+      m[71] = 0x22;
+      m[72] = 0x2f;
+      m[73] = 0xcb;
+      m[74] = 0x88;
+      m[75] = 0x1a;
+      m[76] = 0x4b;
+      m[77] = 0x35;
+      m[78] = 0xc2;
+      m[79] = 0x90;
+      m[80] = 0xa0;
+      m[81] = 0x93;
+      m[82] = 0xac;
+      m[83] = 0x15;
+      m[84] = 0x9c;
+      m[85] = 0xe1;
+      m[86] = 0x34;
+      m[87] = 0x09;
+      m[88] = 0x11;
+      m[89] = 0x1f;
+      m[90] = 0xf0;
+      m[91] = 0x35;
+      m[92] = 0x84;
+     
```

Comment:
you need to return a value here -- right now, all of your tests are designed to print "success" or "failure". this is good for debugging, but we need tests that will fail on CI when something goes wrong. please change every single function to return a boolean indicating success, then change main to look like this:
```
int main () {
  if (!testEcdsaSignature1())
    return EXIT_FAILURE;
  if (!testEcdsaSignature2())
    return EXIT_FAILURE;
  ...

  return EXIT_SUCCESS;
}
```

Reply:
Done. But this test needs to be rewritten in F*.